src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java

```diff
@@ -988,11 +990,11 @@ else if ((entity instanceof WitherEntity) ||
      * @param pos Block position in world
      * @param start The start vector
      * @param end The end vector
-     * @param original The original result from {@link Block#collisionRayTrace(IBlockState, World, BlockPos, Vec3d, Vec3d)}
+     * @param original The original result from {@link IBlockReader#func_217296_a(Vec3d, Vec3d, BlockPos, VoxelShape, BlockState)}
      * @return A result that suits your block
      */
     @Nullable
-    default RayTraceResult getRayTraceResult(BlockState state, World world, BlockPos pos, Vec3d start, Vec3d end, RayTraceResult original)
+    default BlockRayTraceResult getRayTraceResult(BlockState state, IBlockReader world, BlockPos pos, Vec3d start, Vec3d end, BlockRayTraceResult original)
```

Comment:
We do not allow breaking changes, but I fear this one is neccessary, as we cannot safly upcast the IBlockReader and RayTraceResult. Also, mods that simply override this function will not break, and mods should not call this function manually...

Reply:
> mods that simply override this function will not break
they will break if they call super.
So the method should ideally be kept and deprecated.

Reply:
yeah right, forgot it was defaulted...